include/stl2/detail/range/range.hpp

```diff
@@ -23,9 +23,35 @@
 
 STL2_OPEN_NAMESPACE {
 	namespace ext {
-		// iterator_range? view? simple_iterator_and_sentinel_pair?
-		template <Iterator I, Sentinel<I> S = I>
-		using range = tagged_pair<tag::begin(I), tag::end(S)>;
+		// TODO: replace with subrange from P0789
+		template <ext::DestructibleObject I, ext::DestructibleObject S = I>
+		struct range;
+
+		template <Iterator I, Sentinel<I> S>
+		struct range<I, S>
+		: tagged_pair<tag::begin(I), tag::end(S)> {
+			range() = default;
+			using tagged_pair<tag::begin(I), tag::end(S)>::tagged_pair;
+		};
+
+		template <Iterator I, Sentinel<I> S>
+		struct range<dangling<I>, dangling<S>>
```

Comment:
Why prefer `range<dangling<I>, dangling<S>>` to `dangling<range<I, S>>`? Specifically, I note that the former is not a range, which I find bothersome.